src/main/java/app/bpartners/api/file/FileDownloaderImpl.java

```diff
@@ -142,7 +142,8 @@ private static File createFileFrom(String filename, byte[] response) throws IOEx
         || filename.contains("\\")) {
       throw new IllegalArgumentException("Invalid filename");
     }
-    File res = File.createTempFile(filename, null);
+    Path tempDir = Files.createTempDirectory("safe_temp");
```

Comment:
Do not use magic values, put them in static or local variables : `safe_temp`, `null`